src/internal_macros.rs

```diff
@@ -117,21 +117,69 @@ macro_rules! display_from_debug {
     };
 }
 pub(crate) use display_from_debug;
-
-#[cfg(test)]
-macro_rules! hex_script (($s:expr) => (<$crate::Script as core::str::FromStr>::from_str($s).unwrap()));
 #[cfg(test)]
-pub(crate) use hex_script;
+pub(crate) use test_macros::*;
 
 #[cfg(test)]
-macro_rules! hex_hash (($h:ident, $s:expr) => ($h::from_slice(&<$crate::prelude::Vec<u8> as $crate::hashes::hex::FromHex>::from_hex($s).unwrap()).unwrap()));
-#[cfg(test)]
-pub(crate) use hex_hash;
+mod test_macros {
+    use crate::hashes::hex::FromHex;
+    use crate::PublicKey;
+
+    /// Trait used to create a value from hex string for testing purposes.
+    pub(crate) trait TestFromHex {
+        /// Produces the value from hex.
+        ///
+        /// ## Panics
+        ///
+        /// The function panics if the hex or the value is invalid.
+        fn test_from_hex(hex: &str) -> Self;
+    }
 
-#[cfg(test)]
-macro_rules! hex_decode (($h:ident, $s:expr) => (deserialize::<$h>(&<$crate::prelude::Vec<u8> as $crate::hashes::hex::FromHex>::from_hex($s).unwrap()).unwrap()));
-#[cfg(test)]
-pub(crate) use hex_decode;
+    impl<T: FromHex> TestFromHex for T {
+        fn test_from_hex(hex: &str) -> Self { Self::from_hex(hex).unwrap() }
+    }
+
+    impl TestFromHex for PublicKey {
+        fn test_from_hex(hex: &str) -> Self {
+            PublicKey::from_slice(&Vec::from_hex(hex).unwrap()).unwrap()
+        }
+    }
+
+    macro_rules! hex (($hex:literal) => (Vec::from_hex($hex).unwrap()));
+    pub(crate) use hex;
+
+    macro_rules! hex_into {
+        ($hex:expr) => {
+            $crate::internal_macros::hex_into!(_, $hex)
+        };
+        ($type:ty, $hex:expr) => {
+            <$type as $crate::internal_macros::TestFromHex>::test_from_hex($hex)
+        };
+    }
+    pub(crate) use hex_into;
```

Comment:
I think we could get rid of `hex!` altogether and re-name `hex_into!` to `hex!`.

Reply:
I think it's a useful building block for other macros. There's even a case when it's useful in non-test code after it's changed to array instead of vec.

Reply:
no worries.